packages/flutter/lib/src/material/toggleable.dart

```diff
@@ -241,7 +269,7 @@ abstract class RenderToggleable extends RenderConstrainedBox {
 
   void _handleTap() {
     if (isInteractive)
-      onChanged(!_value);
+      onChanged(value == false);
```

Comment:
if tristate is true, this doesn't seem to give the user a way to get back to the indeterminate state. It seems like if tristate is true, this should go in the loop false -> true -> indeterminate.

Reply:
That was intentional however I see that HTML tristates do cycle this way. So we can too.

Reply:
The main reason to do it this way is that if someone _doesn't_ want this behaviour, they can easily just set tristate to false once the value isn't null, but if they _do_ want this behaviour, there's nothing they can do if we don't implement it by default.